python/dgl/graphbolt/itemset.py

```diff
@@ -255,6 +267,44 @@ def __iter__(self) -> Iterator:
     def __len__(self) -> int:
         return sum(len(itemset) for itemset in self._itemsets.values())
 
+    def __getitem__(self, idx: Union[int, slice, Iterable]) -> Dict[str, Tuple]:
+        try:
+            len(self)
+        except TypeError:
+            raise TypeError(
+                f"{type(self).__name__} instance doesn't support indexing."
+            )
+        offsets = [len(itemset) for itemset in self._itemsets.values()]
+        total_num = sum(offsets)
+        if isinstance(idx, int):
+            if idx < 0:
+                idx += total_num
+            assert idx < total_num, (
+                f"Index {idx} out of range for {type(self).__name__} "
+                f"instance with length {total_num}."
+            )
+            for key, offset in zip(self._itemsets.keys(), offsets):
+                if idx < offset:
+                    return {key: self._itemsets[key][idx]}
+                idx -= offset
+        elif isinstance(idx, slice):
+            start, stop, step = idx.indices(total_num)
+            data = {}
+            for key, offset in zip(self._itemsets.keys(), offsets):
+                if start < offset:
```

Comment:
what if start is within a type and stop is within another type?

Reply:
it's supported and covered in testcase.